### PR TITLE
docs: update PR template to consider tests

### PR DESCRIPTION
**Pull request by odockal**
### What does this PR do?
Updates PR template to also consider and recall an author to include unit/component tests in a PR. I have noticed that sometimes people are adding unit/component tests after PR is created. This "Have you added the tests" section in the PR's description could serve as a reminder before submitting it, or provide a place to add ie. a link to another PR/issue that addresses this.

### What issues does this PR fix or reference?
#5998

### CODE REVIEW

**Review by deboer-tim — COMMENTED** [review]
Same comment as @benoitf in #5999: the new section sounds redundant and it's not really clear what the difference is from the existing one. We already require tests, maybe we could just improve the existing comment?

**Comment by odockal** [comment]
@deboer-tim better?

**Review by deboer-tim — APPROVED** [review]

**Review by benoitf — APPROVED** [review]
we could have checkboxes (can be follow up)

like

- [x] Tests are covering the bug-fix or the new feature

**Comment by odockal** [comment]
> we could have checkboxes (can be follow up)
> 
> like
> 
>     * [x]  Tests are covering the bug-fix or the new feature



Great idea, let's do it right away. I assume that checkbox will be unchecked by default, right?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -2,14 +2,17 @@
...
+<!-- Include any related issues from Podman Desktop
 repository (or from another issue tracker). -->
 
 ### How to test this PR?
 
-<!-- Please explain steps to reproduce -->
+<!-- Please explain steps to verify the functionality,
+do not forget to provide unit/component tests -->
+
+- [ ] Tests are covering the bug-fix or the new feature
```
Comment (benoitf) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
- [ ] Tests are covering the bug fix or the new feature
```

**Review by benoitf — COMMENTED** [review]
correcting myself :-)

**Pull request merged**